apps/api/src/routes/v1/dashboardData.ts

```diff
@@ -144,25 +150,34 @@ const route = createRoute({
 });
 
 export default new OpenAPIHono().openapi(route, async c => {
-  const chainStatsQuery = await getChainStats();
-  const dashboardData = await getDashboardData();
-  const networkCapacity = await getNetworkCapacity();
-  const networkCapacityStats = await getProviderGraphData("count");
-  const latestBlocks = await getBlocks(5);
-  const latestTransactions = await getTransactions(5);
-
+  const chainStatsQuery = await runOrLog(
+    getChainStats,
+    {
+      bondedTokens: undefined,
+      totalSupply: undefined,
+      communityPool: undefined,
+      inflation: undefined,
+      stakingAPR: undefined
+    }
+  );
+  const { now, compare } = await runOrLog(getDashboardData)
+  const networkCapacity = await runOrLog(getNetworkCapacity);
+  const networkCapacityStats = await runOrLog(() => getProviderGraphData("count"));
+  const latestBlocks = await runOrLog(() => getBlocks(5));
   const chainStats = {
-    height: latestBlocks[0].height,
-    transactionCount: latestBlocks[0].totalTransactionCount,
-    ...chainStatsQuery
-  };
+    ...chainStatsQuery,
+    height: latestBlocks && latestBlocks.length > 0 ? latestBlocks[0].height : undefined,
+    transactionCount: latestBlocks && latestBlocks.length > 0 ? latestBlocks[0].totalTransactionCount : undefined,
+  }
+  const latestTransactions = await runOrLog(() => getTransactions(5));
```

Comment:
what do you think about one last improvement here? Promise.all should speed up this endpoint quite a bit

```typescript
const [{ now, compare }, chainStatsQuery, networkCapacity, networkCapacityStats, latestBlocks, latestTransactions] = await Promise.all([
    runOrLog(getDashboardData),
    runOrLog(getChainStats, {
      bondedTokens: undefined,
      totalSupply: undefined,
      communityPool: undefined,
      inflation: undefined,
      stakingAPR: undefined
    }),
    runOrLog(getNetworkCapacity),
    runOrLog(() => getProviderGraphData("count")),
    runOrLog(() => getBlocks(5)),
    runOrLog(() => getTransactions(5))
  ]);

  const chainStats = {
    ...chainStatsQuery,
    height: latestBlocks && latestBlocks.length > 0 ? latestBlocks[0].height : undefined,
    transactionCount: latestBlocks && latestBlocks.length > 0 ? latestBlocks[0].totalTransactionCount : undefined
  };
```